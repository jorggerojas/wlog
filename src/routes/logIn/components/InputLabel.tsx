// @flow
import React from "react";
import { Input } from "../../../styles/styles";

interface InputProps {
  label: string;
  forTag: string;
  props: any;
  type: string;
  icon: string;
  placeholder?: string;
  id?: string;
  labelDirection?: string;
}

const InputLabel = ({
  label,
  forTag,
  props,
  type,
  icon,
  placeholder,
  id,
  labelDirection,
}: InputProps) => {
  return (
    <div>
      <label
        className={`uk-form-label uk-text-${
          labelDirection ?? "left"
        } salmonSpanBold`}
        htmlFor={id ? forTag : ""}
      >
        {label}
      </label>
      <div className="uk-inline uk-flex">
        <div className="uk-form-controls">
          <span
            className="uk-form-icon uk-text-bolder salmonSpan"
            uk-icon={`icon: ${icon}`}
          />
          <Input
            className={`uk-text-large uk-input uk-form-width-large`}
            id={id ?? ""}
            type={type}
            placeholder={placeholder}
            {...props}
          />
        </div>
      </div>
    </div>
  );
};

export default InputLabel;
