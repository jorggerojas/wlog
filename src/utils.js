import React from "react";
import { render, waitFor } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router, Route } from "react-router-dom";
import "@testing-library/jest-dom";

const renderWithRouter = (ui, { route = "/", ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  const finishLoading = () =>
    waitFor(() =>
      expect(utils.queryByText("Loading, please wait...")).toBeNull()
    );
  return {
    ...utils,
    finishLoading,
    history,
  };
};

export {
  Simulate,
  wait,
  render,
  cleanup,
  renderIntoDocument,
  fireEvent,
} from "@testing-library/react";
export { renderWithRouter };

export function renderWithProviders(
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
  apolloMocks
) {
  console.log("I am route inside renderWithProviders Wrapper:", route);
  return {
    ...render(
      <Router history={history}>
        <Route path={route}>{ui}</Route>
      </Router>
    ),
    history,
  };
}
