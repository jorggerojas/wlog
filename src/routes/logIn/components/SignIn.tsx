// @flow
import React from "react";
import { Link } from "react-router-dom";
import { useFormik } from "formik";
import cookie from "react-cookies";
import * as yup from "yup";
import axios from "axios";
import swal from "sweetalert";
import InputLabel from "./InputLabel";
import { URL } from "../../../config";
interface SignInProps {
  handle: Function;
  loading: Function;
}

const SignIn = ({ handle, loading }: SignInProps) => {
  const info = (username: string) => axios.get(`${URL}/users/${username}`);
  const data = (username: string, pass: string) => {
    axios
      .post(`${URL}/login`, {
        nickname: username,
        pass: pass,
      })
      .then(({ headers }) => {
        setTimeout(() => {
          info(username).then(({ data }) => {
            if (data.isBlocked === "0") {
              cookie.save("TOKEN", headers.authorization.split(" ")[1], {
                path: "/",
              });
              cookie.save("USER", data.nickname, { path: "/" });
              cookie.save("ROLE", data.role[0], { path: "/" });
              loading(false);
              window.location.href = "/";
            } else {
              swal({
                title: "Hey...",
                text: "We're sorry. Your account has been blocked",
                icon: "warning",
                buttons: {
                  cancel: { visible: false },
                  ok: {
                    className: "uk-button uk-button-warning ",
                    text: "Ok",
                    visible: true,
                  },
                },
                className: "uk-card",
                dangerMode: true,
              }).then(() => {
                loading(false);
              });
            }
          });
        }, 500);
      })
      .catch(() => {
        loading(false);
        swal({
          title: "Ooops...",
          text: "We couldn't find the user",
          icon: "error",
          buttons: {
            ok: { visible: false },
            cancel: {
              className: "uk-button uk-button-danger ",
              text: "Ok",
              visible: true,
            },
          },
          className: "uk-card",
          dangerMode: true,
        });
      });
  };
  const formik = useFormik({
    initialValues: {
      username: "",
      pass: "",
    },
    validationSchema: yup.object({
      username: yup
        .string()
        .required("Username is required")
        .max(15, "Username must be at most 15 characters")
        .min(4, "Username must be at least 5 characters"),
      pass: yup
        .string()
        .required("Password is required")
        .min(5, "Password must be at least 5 characters"),
    }),
    onSubmit: function ({ username, pass }) {
      loading(true);
      data(username, pass);
    },
  });
  return (
    <div
      className="uk-width-1-1 uk-height-large uk-text-right uk-animation-slide-right"
      uk-height-viewport="expand: true"
    >
      <h1 className=" uk-padding-small uk-margin-remove uk-margin-bottom">
        <Link
          to="/"
          className="uk-button-text"
          style={{
            color: "#fd7777",
            fontWeight: 500,
            letterSpacing: 3,
            textDecoration: "none",
          }}
        >
          WLOG{" "}
        </Link>
        <span className="uk-text-light"> - Sign In</span>
      </h1>
      <form
        method="POST"
        className="uk-grid-small uk-form-stacked uk-padding-small "
        uk-grid=""
        onSubmit={formik.handleSubmit}
      >
        <div className="uk-margin uk-width-3-4@s uk-margin-small ">
          <InputLabel
            forTag="username"
            placeholder="Your username"
            id="username"
            type="text"
            label="Username"
            icon="user"
            props={{ ...formik.getFieldProps("username") }}
          />
          {formik.touched.username && formik.errors.username ? (
            <div className="uk-text-danger uk-text-bold">
              {formik.errors.username}
            </div>
          ) : null}
        </div>
        <div className="uk-margin uk-width-3-4@s uk-margin-small">
          <InputLabel
            forTag="pass"
            placeholder="Your password"
            id="pass"
            type="password"
            label="Password"
            icon="lock"
            props={{ ...formik.getFieldProps("pass") }}
          />
          {formik.touched.pass && formik.errors.pass ? (
            <div className="uk-text-danger uk-text-bold">
              {formik.errors.pass}
            </div>
          ) : null}
        </div>
        <div className="uk-margin-top uk-width-3-4@s uk-margin">
          <input
            className="uk-button uk-button-primary"
            type="submit"
            value="Sign In"
          />
        </div>
      </form>
      <p className="uk-padding-small">
        Don't have an account?{" "}
        <span
          onClick={() => {
            handle();
          }}
        >
          <Link to="#">Sign up</Link>
        </span>
      </p>
    </div>
  );
};

export default SignIn;
