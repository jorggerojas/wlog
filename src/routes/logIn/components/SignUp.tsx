// @flow
import React from "react";
import { Link } from "react-router-dom";
import { useFormik } from "formik";
import * as yup from "yup";
import axios from "axios";
import swal from "sweetalert";
import InputLabel from "./InputLabel";
import { URL } from "../../../config";
import anime from "animejs";
import t from "typy";
import {
  LogoSignIn,
  TextChange,
  SpanLight,
  LinkUserComment,
  Submit,
} from "../../../styles/text";

interface SignUpProps {
  handle: Function;
  loading: Function;
}

const SignUp = ({ handle, loading }: SignUpProps) => {
  const sign = ({ fullName, mail, username, passConfirm, pass }: any) => {
    axios
      .post(`${URL}/users`, {
        name: fullName,
        dateLog: null,
        mail: mail,
        pass: pass,
        role: ["LECTOR"],
        isBlocked: 0,
        colorTheme: null,
        suscriptionUser: [],
        suscriptionTheme: [],
        nickname: username,
      })
      .then(({ data }: any) => {
        swal({
          title: "Success",
          text: "Just log in and enjoy!",
          icon: "success",
          buttons: {
            cancel: { visible: false },
            ok: {
              className: "uk-button uk-button-success ",
              text: "Let's do it",
              visible: true,
            },
          },
          className: "uk-card",
          dangerMode: false,
        }).then(() => {
          loading(false);
          handle();
        });
      })
      .catch((error) => {
        loading(false);
        if (error.response.status === 409) {
          swal({
            title: "Ooops...",
            text: "The username is already used",
            icon: "warning",
            buttons: {
              ok: { visible: false },
              cancel: {
                className: "uk-button uk-button-danger ",
                text: "Ok",
                visible: true,
              },
            },
            className: "uk-card",
            dangerMode: true,
          });
        }
      });
  };
  const animation = anime({
    targets: "input[type='text'], input[type='password'], input[type='mail']",
    translateX: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 14, 16, 18, 20, 0].map(
      (num: Number) => num
    ),
    duration: 100,
    loop: 2,
    autoplay: false,
    direction: "reverse",
  });
  const formik = useFormik({
    initialValues: {
      fullName: "",
      mail: "",
      username: "",
      pass: "",
      passConfirm: "",
    },
    validationSchema: yup.object({
      fullName: yup
        .string()
        .required("Full name is required")
        .min(5, "Full name must be at least 5 characters")
        .max(50, "Full name must be at most 50 characters"),
      mail: yup
        .string()
        .required("Email is required")
        .email("Email must be a valid email"),
      username: yup
        .string()
        .required("Username is required")
        .min(4, "Username must be at least 4 characters")
        .max(15, "Username must be at most 15 characters"),
      pass: yup
        .string()
        .required("Password is required")
        .min(5, "Password must be at least 5 characters"),
      passConfirm: yup
        .string()
        .required("Confirm password is required")
        .oneOf([yup.ref("pass")], "Passwords must be the same"),
    }),
    onSubmit: (values) => {
      loading(true);
      sign(values);
    },
  });
  return (
    <div className="uk-overflow-auto uk-height-large uk-width-1-1 uk-text-right uk-animation-slide-bottom">
      <h1 className=" uk-padding-small uk-margin-remove uk-margin-bottom">
        <Link to="/" className="uk-button-text uk-link-reset">
          <LogoSignIn>WLOG</LogoSignIn>
        </Link>
        <SpanLight className="uk-text-light"> - Sign Up</SpanLight>
      </h1>
      <form
        className="uk-grid uk-form-stacked uk-padding-small "
        uk-grid=""
        onSubmit={formik.handleSubmit}
      >
        <div className="uk-width-3-4@s uk-margin-small ">
          <InputLabel
            forTag="fullName"
            type="text"
            icon="happy"
            label="Full name"
            id="fullName"
            placeholder="Your full name"
            props={{ ...formik.getFieldProps("fullName") }}
          />
          {t(formik.touched.fullName).isTrue &&
          !t(formik.errors.fullName).isNullOrUndefined ? (
            <div className="uk-text-danger uk-text-bold uk-text-left">
              {formik.errors.fullName}
              {!t(formik.errors.fullName).isNullOrUndefined
                ? animation.play()
                : animation.pause()}
            </div>
          ) : null}
        </div>
        <div className="uk-width-3-4@s uk-margin-small">
          <InputLabel
            forTag="mail"
            type="email"
            icon="mail"
            label="Email"
            id="mail"
            placeholder="Your full email"
            props={{ ...formik.getFieldProps("mail") }}
          />
          {t(formik.touched.mail).isTrue &&
          !t(formik.errors.mail).isNullOrUndefined ? (
            <div className="uk-text-danger uk-text-bold uk-text-left">
              {formik.errors.mail}
              {!t(formik.errors.mail).isNullOrUndefined
                ? animation.play()
                : animation.pause()}
            </div>
          ) : null}
        </div>
        <div className="uk-width-3-4@s uk-margin-small ">
          <InputLabel
            forTag="username"
            placeholder="Your username"
            id="username"
            type="text"
            label="Username"
            icon="user"
            props={{ ...formik.getFieldProps("username") }}
          />
          {t(formik.touched.username).isTrue &&
          !t(formik.errors.username).isNullOrUndefined ? (
            <div className="uk-text-danger uk-text-bold uk-text-left">
              {formik.errors.username}
              {!t(formik.errors.username).isNullOrUndefined
                ? animation.play()
                : animation.pause()}
            </div>
          ) : null}
        </div>
        <div className="uk-margin uk-width-3-4@s uk-margin-small">
          <InputLabel
            forTag="pass"
            placeholder="Your password"
            id="pass"
            type="password"
            label="Password"
            icon="lock"
            props={{ ...formik.getFieldProps("pass") }}
          />
          {t(formik.touched.pass).isTrue &&
          !t(formik.errors.pass).isNullOrUndefined ? (
            <div className="uk-text-danger uk-text-bold uk-text-left">
              {formik.errors.pass}
              {!t(formik.errors.pass).isNullOrUndefined
                ? animation.play()
                : animation.pause()}
            </div>
          ) : null}
        </div>
        <div className="uk-margin uk-width-3-4@s uk-margin-small">
          <InputLabel
            forTag="passConfirm"
            placeholder="Confirm your password"
            id="passConfirm"
            type="password"
            label="Confirm your password"
            icon="lock"
            props={{ ...formik.getFieldProps("passConfirm") }}
          />
          {t(formik.touched.passConfirm).isTrue &&
          !t(formik.errors.passConfirm).isNullOrUndefined ? (
            <div className="uk-text-danger uk-text-bold uk-text-left">
              {formik.errors.passConfirm}
              {!t(formik.errors.passConfirm).isNullOrUndefined
                ? animation.play()
                : animation.pause()}
            </div>
          ) : null}
        </div>
        <div className="uk-margin-top uk-width-4-4@s uk-margin uk-text-left">
          <Submit
            className="uk-button uk-button-primary"
            type="submit"
            value="Sign Up"
          />
        </div>
      </form>
      <TextChange className="uk-padding-small">
        Already a Wlog user?{" "}
        <span
          onClick={() => {
            handle();
          }}
        >
          <LinkUserComment href="#s">Sign in</LinkUserComment>
        </span>
      </TextChange>
    </div>
  );
};

export default SignUp;
