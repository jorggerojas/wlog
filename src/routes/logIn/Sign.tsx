import React from "react";
import SignIn from "./components/SignIn";
import SignUp from "./components/SignUp";

const Sign = () => {
  return (
    <div
      style={{
        background: "#fd7777 no-repeat",
      }}
    >
      <div
        style={{
          background: "no-repeat",
          backgroundSize: "cover",
          mixBlendMode: "lighten",
          opacity: 0.5,
          filter: "blur(1.4px)",
        }}
        className=" uk-cover-container uk-width-5-5@s uk-height-medium uk-flex uk-flex-center uk-flex-middle uk-background-cover uk-light"
        data-src="https://www.wallpapertip.com/wmimgs/219-2191818_os-x-mavericks-activity-monitor-change-the-login.jpg"
        uk-img=""
        uk-height-viewport=""
      ></div>
      <div
        // style={{ backgroundColor: "#fd7777" }}
        className="uk-position-large uk-position-cover uk-overlay uk-position-z-index uk-flex uk-flex-center uk-flex-middle"
      >
        <SignIn />
        <SignUp />
      </div>
    </div>
  );
};

export default Sign;
