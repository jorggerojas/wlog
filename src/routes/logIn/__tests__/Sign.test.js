import React from "react";
import { cleanup, fireEvent, waitFor } from "@testing-library/react";
import axiosMock from "axios";
import Sign from "../Sign";
import InputLabel from "../components/InputLabel";
import SignIn from "../components/SignIn";
import SignUp from "../components/SignUp";
import { renderWithRouter } from "../../../utils";

afterEach(cleanup);
jest.mock("axios");

describe("<Sign/> tree", () => {
  test("render with <SignIn/> component", async () => {
    const { container } = renderWithRouter(<Sign />, "/sign");
    await expect(container).toHaveTextContent("Sign In");
    await expect(container).toMatchSnapshot();
  });
  test("render with <SignUp/> component", async () => {
    const { container, findByText } = renderWithRouter(<Sign />, "/sign");
    const change = await findByText("Sign up");
    await expect(container).toHaveTextContent("Sign In");
    await fireEvent.click(change);
    await expect(container).toHaveTextContent("Sign Up");
    await expect(container).toHaveTextContent("Full name");
    await expect(container).toMatchSnapshot();
  });
  test("render <InputLabel/> component", () => {
    const props = {
      label: "RealInput",
      forTag: "realInput",
      props: {},
      type: "text",
      icon: "user",
      placeholder: "My Input",
      id: "realInput",
      labelDirection: "",
    };
    const { container, getByPlaceholderText } = renderWithRouter(
      <InputLabel {...props} />
    );
    const input = getByPlaceholderText(props.placeholder);
    expect(container).toHaveTextContent(props.label);
    expect(container).toBeInTheDocument(input);
    expect(container).toMatchSnapshot();
  });
});
