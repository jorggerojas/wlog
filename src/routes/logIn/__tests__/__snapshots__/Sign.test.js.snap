// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sign/> tree render <InputLabel/> component 1`] = `
<div>
  <div>
    <label
      class="uk-form-label uk-text- salmonSpanBold"
      for="realInput"
    >
      RealInput
    </label>
    <div
      class="uk-inline uk-flex"
    >
      <div
        class="uk-form-controls"
      >
        <span
          class="uk-form-icon uk-text-bolder salmonSpan"
          uk-icon="icon: user"
        />
        <input
          class="sc-bkzZxe gzgMFm uk-text-large uk-input uk-form-width-large"
          data-testid="realInput"
          id="realInput"
          name="realInput"
          placeholder="My Input"
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Sign/> tree render with <SignIn/> component 1`] = `
<div>
  <div
    class="sc-hiSbYr bSxRoH uk-grid uk-animation-fade"
  >
    <div
      class="sc-bYEvPH cLJKVE uk-width-1-1 uk-hidden uk-animation-fade uk-margin-large-top uk-position-absolute uk-position-z-index uk-position-large
        uk-position-center uk-overlay uk-overlay-default uk-animation-fade"
    >
      <h3
        class="sc-pFZIQ hNoSbE uk-text-center uk-margin-large-top uk-text-bold"
      >
        Loading, please wait...
      </h3>
      <div
        class="uk-text-center uk-margin"
      >
        <img
          alt="Loading"
          class="sc-iqHYGH fKosLb"
          src="load.svg"
        />
      </div>
    </div>
    <div
      class=" uk-width-2-5@s uk-visible@s uk-text-center uk-text-middle uk-flex uk-flex-middle uk-flex-center"
    >
      <a
        class="uk-link-reset"
        href="/"
      >
        <h1
          class="sc-crrsfI fKYnZb uk-text-center uk-text-middle uk-text-bold"
        >
          WLOG
        </h1>
      </a>
    </div>
    <div
      class="sc-gWHgXt CLsWb  uk-section-default uk-width-3-5@s uk-flex-middle uk-flex uk-flex-center uk-margin uk-padding"
    >
      <div
        class="uk-width-1-1 uk-height-large uk-text-right uk-animation-slide-right"
        uk-height-viewport="expand: true"
      >
        <h1
          class=" uk-padding-small uk-margin-remove uk-margin-bottom"
        >
          <a
            class="uk-button-text uk-link-reset"
            href="/"
          >
            <span
              class="sc-bqyKva hfLAAC"
            >
              WLOG
            </span>
          </a>
          <span
            class="sc-fodVxV cGXbwS uk-text-light"
          >
             - Sign In
          </span>
        </h1>
        <form
          class="uk-grid-small uk-form-stacked uk-padding-small "
          method="POST"
          uk-grid=""
        >
          <div
            class="uk-margin uk-width-3-4@s uk-margin-small "
          >
            <div>
              <label
                class="uk-form-label uk-text-left salmonSpanBold"
                for="username"
              >
                Username
              </label>
              <div
                class="uk-inline uk-flex"
              >
                <div
                  class="uk-form-controls"
                >
                  <span
                    class="uk-form-icon uk-text-bolder salmonSpan"
                    uk-icon="icon: user"
                  />
                  <input
                    class="sc-bkzZxe gzgMFm uk-text-large uk-input uk-form-width-large"
                    data-testid="username"
                    id="username"
                    name="username"
                    placeholder="Your username"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="uk-margin uk-width-3-4@s uk-margin-small"
          >
            <div>
              <label
                class="uk-form-label uk-text-left salmonSpanBold"
                for="pass"
              >
                Password
              </label>
              <div
                class="uk-inline uk-flex"
              >
                <div
                  class="uk-form-controls"
                >
                  <span
                    class="uk-form-icon uk-text-bolder salmonSpan"
                    uk-icon="icon: lock"
                  />
                  <input
                    class="sc-bkzZxe gzgMFm uk-text-large uk-input uk-form-width-large"
                    data-testid="pass"
                    id="pass"
                    name="pass"
                    placeholder="Your password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="uk-margin-top uk-width-3-4@s uk-margin"
          >
            <input
              class="sc-dmlrTW Iqyzn uk-button uk-button-primary"
              type="submit"
              value="Sign In"
            />
          </div>
        </form>
        <p
          class="sc-kstrdz eQcRWX uk-padding-small"
        >
          Don't have an account?
           
          <span>
            <a
              class="sc-hKgILt evLhsB"
              href="#s"
            >
              Sign up
            </a>
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Sign/> tree render with <SignUp/> component 1`] = `
<div>
  <div
    class="sc-hiSbYr bSxRoH uk-grid uk-animation-fade"
  >
    <div
      class="sc-bYEvPH cLJKVE uk-width-1-1 uk-hidden uk-animation-fade uk-margin-large-top uk-position-absolute uk-position-z-index uk-position-large
        uk-position-center uk-overlay uk-overlay-default uk-animation-fade"
    >
      <h3
        class="sc-pFZIQ hNoSbE uk-text-center uk-margin-large-top uk-text-bold"
      >
        Loading, please wait...
      </h3>
      <div
        class="uk-text-center uk-margin"
      >
        <img
          alt="Loading"
          class="sc-iqHYGH fKosLb"
          src="load.svg"
        />
      </div>
    </div>
    <div
      class=" uk-width-2-5@s uk-visible@s uk-text-center uk-text-middle uk-flex uk-flex-middle uk-flex-center"
    >
      <a
        class="uk-link-reset"
        href="/"
      >
        <h1
          class="sc-crrsfI fKYnZb uk-text-center uk-text-middle uk-text-bold"
        >
          WLOG
        </h1>
      </a>
    </div>
    <div
      class="sc-gWHgXt CLsWb  uk-section-default uk-width-3-5@s uk-flex-middle uk-flex uk-flex-center uk-margin uk-padding"
    >
      <div
        class="uk-overflow-auto uk-height-large uk-width-1-1 uk-text-right uk-animation-slide-bottom"
      >
        <h1
          class=" uk-padding-small uk-margin-remove uk-margin-bottom"
        >
          <a
            class="uk-button-text uk-link-reset"
            href="/"
          >
            <span
              class="sc-bqyKva hfLAAC"
            >
              WLOG
            </span>
          </a>
          <span
            class="sc-fodVxV cGXbwS uk-text-light"
          >
             - Sign Up
          </span>
        </h1>
        <form
          class="uk-grid uk-form-stacked uk-padding-small "
          uk-grid=""
        >
          <div
            class="uk-width-3-4@s uk-margin-small "
          >
            <div>
              <label
                class="uk-form-label uk-text-right salmonSpanBold"
                for="fullName"
              >
                Full name
              </label>
              <div
                class="uk-inline uk-flex"
              >
                <div
                  class="uk-form-controls"
                >
                  <span
                    class="uk-form-icon uk-text-bolder salmonSpan"
                    uk-icon="icon: happy"
                  />
                  <input
                    class="sc-bkzZxe gzgMFm uk-text-large uk-input uk-form-width-large"
                    data-testid="fullName"
                    id="fullName"
                    name="fullName"
                    placeholder="Your full name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="uk-width-3-4@s uk-margin-small"
          >
            <div>
              <label
                class="uk-form-label uk-text-right salmonSpanBold"
                for="mail"
              >
                Email
              </label>
              <div
                class="uk-inline uk-flex"
              >
                <div
                  class="uk-form-controls"
                >
                  <span
                    class="uk-form-icon uk-text-bolder salmonSpan"
                    uk-icon="icon: mail"
                  />
                  <input
                    class="sc-bkzZxe gzgMFm uk-text-large uk-input uk-form-width-large"
                    data-testid="mail"
                    id="mail"
                    name="mail"
                    placeholder="Your full email"
                    type="email"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="uk-width-3-4@s uk-margin-small "
          >
            <div>
              <label
                class="uk-form-label uk-text-right salmonSpanBold"
                for="username"
              >
                Username
              </label>
              <div
                class="uk-inline uk-flex"
              >
                <div
                  class="uk-form-controls"
                >
                  <span
                    class="uk-form-icon uk-text-bolder salmonSpan"
                    uk-icon="icon: user"
                  />
                  <input
                    class="sc-bkzZxe gzgMFm uk-text-large uk-input uk-form-width-large"
                    data-testid="username"
                    id="username"
                    name="username"
                    placeholder="Your username"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="uk-margin uk-width-3-4@s uk-margin-small"
          >
            <div>
              <label
                class="uk-form-label uk-text-right salmonSpanBold"
                for="pass"
              >
                Password
              </label>
              <div
                class="uk-inline uk-flex"
              >
                <div
                  class="uk-form-controls"
                >
                  <span
                    class="uk-form-icon uk-text-bolder salmonSpan"
                    uk-icon="icon: lock"
                  />
                  <input
                    class="sc-bkzZxe gzgMFm uk-text-large uk-input uk-form-width-large"
                    data-testid="pass"
                    id="pass"
                    name="pass"
                    placeholder="Your password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="uk-margin uk-width-3-4@s uk-margin-small"
          >
            <div>
              <label
                class="uk-form-label uk-text-right salmonSpanBold"
                for="passConfirm"
              >
                Confirm your password
              </label>
              <div
                class="uk-inline uk-flex"
              >
                <div
                  class="uk-form-controls"
                >
                  <span
                    class="uk-form-icon uk-text-bolder salmonSpan"
                    uk-icon="icon: lock"
                  />
                  <input
                    class="sc-bkzZxe gzgMFm uk-text-large uk-input uk-form-width-large"
                    data-testid="passConfirm"
                    id="passConfirm"
                    name="passConfirm"
                    placeholder="Confirm your password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="uk-margin-top uk-width-4-4@s uk-margin uk-text-right"
          >
            <input
              class="sc-dmlrTW Iqyzn uk-button uk-button-primary"
              type="submit"
              value="Sign Up"
            />
          </div>
        </form>
        <p
          class="sc-kstrdz eQcRWX uk-padding-small"
        >
          Already a Wlog user?
           
          <span>
            <a
              class="sc-hKgILt evLhsB"
              href="#s"
            >
              Sign in
            </a>
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
